src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,632 @@
+  fromAssetSymbol?: string;
+  amount?: string;
+  accountType: AccountTypes;
+  receiverAddress?: string;
+  offer?: ExchangeOffer;
+}
+
+const Title = styled.h3`
+  margin: 0 0 18px;
+  padding: 0;
+  font-size: 16px;
+  color: ${({ theme }) => theme.color.text.cardTitle};
+  font-family: 'PTRootUIWebBold', sans-serif;
+`;
+
+const WalletReceiveWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const OfferDetails = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  font-family: 'PTRootUIWebMedium', sans-serif;
+`;
+
+const Container = styled.div`
+  align-items: center;
+  font-family: 'PTRootUIWebMedium', sans-serif;
+  background: #21002e;
+  color: #fefefe;
+  padding: 16px;
+  margin: 5px;
+  border-image: linear-gradient(#346ecd, #cd34a2) 30;
+  border-width: 2px;
+  border-style: solid;
+`;
+
+const Value = styled.div`
+  font-family: 'PTRootUIWebMedium', sans-serif;
+  color: #57c2d6;
+  display: contents;
+`;
+
+const Total = styled.div`
+  font-family: 'PTRootUIWebMedium', sans-serif;
+  color: #ff0065;
+  display: contents;
+  font-weight: bold;
+`;
+
+const HorizontalLine = styled.div`
+  margin: 9px 0;
+  width: 100%;
+  height: 2px;
+  background: linear-gradient(90deg, #23a9c9, #cd34a2);
+`;
+
+const Bold = styled.p`
+  font-weight: bold;
+`;
+
+//TODO : change color of block if polygon has less than 10,000 plr on keybased / smart wallet
+const Block = styled.div`
+  font-size: 12px;
+  padding: 2px;
+  text-indent: 2px;
+  display: flex;
+  color: ${(props) => props.color === 'red' && '#ff0065'};
+`;
+
+const mapOfferToOption = (offer: ExchangeOffer) => {
+  const serviceDetails = swapServiceIdToDetails[offer.provider];
+  return {
+    title: serviceDetails?.title ?? offer.provider,
+    value: offer.provider,
+    iconUrl: serviceDetails?.iconUrl,
+  };
+};
+
+// TODO:UNCOMMENT
+const MAX_PLR_TOKEN_LIMIT = 5;
+
+const PlrDaoStakingTransactionBlock = ({
+  id: transactionBlockId,
+  errorMessages,
+  values,
+  multiCallData,
+}: IPlrDaoStakingMembershipBlock) => {
+  const {
+    smartWalletOnly,
+    providerAddress,
+    accountAddress,
+    sdk,
+    getSupportedAssetsWithBalancesForChainId,
+  } = useEtherspot();
+
+  const [amount, setAmount] = useState<string>('');
+  const [selectedOffer, setSelectedOffer] = useState<SelectOption | null>(
+    values?.offer ? mapOfferToOption(values?.offer) : null
+  );
+  const [availableOffers, setAvailableOffers] = useState<
+    ExchangeOffer[] | null
+  >(values?.offer ? [values.offer] : null);
+  const [isLoadingAvailableOffers, setIsLoadingAvailableOffers] =
+    useState<boolean>(false);
+
+  const [selectedFromAsset, setSelectedFromAsset] =
+    useState<IAssetWithBalance | null>(values?.selectedAsset ?? null);
+
+  const [selectedAccountType, setSelectedAccountType] = useState<string>(
+    AccountTypes.Contract
+  );
+
+  const [selectedFromNetwork, setSelectedFromNetwork] = useState<Chain | null>(
+    null
+  );
```

Comment:
please make sure you use consistent line breaking, seems that when it comes to hooks you are switching between break on type vs break on value, while here really line lenght is fine and no break needed at all?
```suggestion
  const [selectedFromNetwork, setSelectedFromNetwork] = useState<Chain | null>(null);
```